bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/ResumingStreamingResultScanner.java

```diff
@@ -87,25 +103,82 @@ public FlatRow next() throws IOException {
       try {
         FlatRow result = currentDelegate.next();
         if (result != null) {
-          retryHandler.update(result);
+          requestManager.updateLastFoundKey(result.getRowKey());
+          // We've had at least one successful RPC, reset the backoff and retry counter
+          currentErrorBackoff = null;
+          timeoutRetryCount = null;
```

Comment:
I dug into this a bit more, and there seems to be another reason for a null timer.  There's a timer in the Backoff object that starts when the ExponentialBackOff is built.  (The code is [here](https://github.com/google/google-http-java-client/blob/dev/google-http-client/src/main/java/com/google/api/client/util/ExponentialBackOff.java#L171)).

Maybe calling reset on the backoff object for every scan result was the costly part?  I don't completely remember the details, but I do recall settling on this.

I think we'd need to do a bit more analysis and performance testing before we reinstate the creation of the backoff outside of exception handling.